File: J_se_les12/HW/Core/Logic/Remove/RemoveBase.java
package Java.J_se_les12.HW.Core.Logic.Remove;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutErrors;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutSuc;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class RemoveBase {
    Phonebook phonebook;
    IConsoleOutSuc outSuc;
    IConsoleOutErrors error;

    protected RemoveBase(Phonebook phonebook, IConsoleOutSuc outSuc, IConsoleOutErrors error) {
        this.outSuc = outSuc;
        this.phonebook = phonebook;
        this.error = error;
    }

    public abstract void remove(int index);
}

File: J_se_les13/HW/Core/Logic/Remove/RemoveBase.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;

namespace HW.Core.Logic.Remove
{
    public abstract class RemoveBase
    {

        public PhonebookV1 Phonebook { get; set; }

        public IConsoleOutSuc OutSuc { get; set; }
        public IConsoleOutError Error { get; set; }
        protected RemoveBase(PhonebookV1 phonebook, IConsoleOutSuc outSuc, IConsoleOutError error)
        {
            Phonebook = phonebook;
            OutSuc = outSuc;
            Error = error;
        }
        public abstract void Remove(int index);
    }
}